src/wp-includes/class-wp-modules.php

```diff
@@ -0,0 +1,251 @@
+			$this->registered[ $module_identifier ]['enqueued'] = false;
+		}
+		unset( $this->enqueued_before_registered[ $module_identifier ] );
+	}
+
+	/**
+	 * Returns the import map array.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return array Array with an `imports` key mapping to an array of module identifiers and their respective URLs,
+	 *               including the version query.
+	 */
+	public function get_import_map() {
+		$imports = array();
+		foreach ( $this->get_dependencies( array_keys( $this->get_enqueued() ) ) as $module_identifier => $module ) {
+			$imports[ $module_identifier ] = $module['src'] . $this->get_version_query_string( $module['version'] );
+		}
+		return array( 'imports' => $imports );
+	}
+
+	/**
+	 * Prints the import map using a script tag with a type="importmap" attribute.
+	 *
+	 * @since 6.5.0
+	 */
+	public function print_import_map() {
+		$import_map = $this->get_import_map();
+		if ( ! empty( $import_map['imports'] ) ) {
+			wp_print_inline_script_tag(
+				wp_json_encode( $import_map, JSON_HEX_TAG | JSON_HEX_AMP ),
+				array(
+					'type' => 'importmap',
+				)
+			);
+		}
+	}
+
+	/**
+	 * Prints all the enqueued modules using script tags with type="module"
+	 * attributes.
+	 *
+	 * @since 6.5.0
+	 */
+	public function print_enqueued_modules() {
+		foreach ( $this->get_enqueued() as $module_identifier => $module ) {
+			wp_print_script_tag(
+				array(
+					'type' => 'module',
+					'src'  => $module['src'] . $this->get_version_query_string( $module['version'] ),
+					'id'   => $module_identifier,
+				)
+			);
+		}
+	}
+
+	/**
+	 * Prints the the static dependencies of the enqueued modules using link tags
+	 * with rel="modulepreload" attributes.
+	 *
+	 * If a module has already been enqueued, it will not be preloaded.
+	 *
+	 * @since 6.5.0
+	 */
+	public function print_module_preloads() {
+		foreach ( $this->get_dependencies( array_keys( $this->get_enqueued() ), array( 'static' ) ) as $module_identifier => $module ) {
+			if ( true !== $module['enqueued'] ) {
+				echo sprintf(
+					'<link rel="modulepreload" href="%s" id="%s">',
+					esc_attr( $module['src'] . $this->get_version_query_string( $module['version'] ) ),
+					esc_attr( $module_identifier )
+				);
+			}
+		}
+	}
+
+	/**
+	 * Gets the version of a module.
+	 *
+	 * If $version is set to false, the version number is the currently installed
+	 * WordPress version. If $version is set to null, no version is added.
+	 * Otherwise, the string passed in $version is used.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string|false|null $version The version of the module.
+	 * @return string A string with the version, prepended by `?ver=`, or an empty string if there is no version.
+	 */
+	private function get_version_query_string( $version ) {
+		if ( false === $version ) {
+			return '?ver=' . get_bloginfo( 'version' );
+		} elseif ( null !== $version ) {
+			return '?ver=' . $version;
+		}
+		return '';
+	}
```

Comment:
```suggestion
	/**
	 * Gets the versioned URL for a module src.
	 *
	 * If $version is set to false, the version number is the currently installed
	 * WordPress version. If $version is set to null, no version is added.
	 * Otherwise, the string passed in $version is used.
	 *
	 * @since 6.5.0
	 *
	 * @param array $module The module.
	 * @return string The module src with a version if relevant.
	 */
	private function get_version_query_string( array $module ) {
		$args = array();
		if ( false === $module['version'] ) {
			$args['ver'] = get_bloginfo( 'version' );
		} elseif ( null !== $module['version'] ) {
			$args['ver'] = $module['version'];
		}
		if ( $args ) {
			return add_query_arg( $args, $module['src'] );
		}
		return $module['src'];
	}
```

Reply:
Makes sense. Thanks, Weston!

Done in https://github.com/WordPress/wordpress-develop/pull/5818/commits/1df672d3da05ae7bfdf52caffac3a9d4d38752e1.